crates/client/src/storage/fs.rs

```diff
@@ -76,7 +79,16 @@ impl FileSystemPackageStorage {
 }
 
 #[async_trait]
-impl PackageStorage for FileSystemPackageStorage {
+impl RegistryStorage for FileSystemPackageStorage {
+    async fn load_checkpoint(&self) -> Result<Option<SerdeEnvelope<MapCheckpoint>>> {
+        let contents = load(&self.base_dir.join("checkpoint")).await;
+        contents
```

Comment:
Nit (as it's not really the contents, just the result):

```suggestion
        load(&self.base_dir.join("checkpoint")).await
```